### PR TITLE
Return error on mlx plugin on firmware reset check

### PR DESCRIPTION
**Pull request by SchSeba**
This commit makes the functions to return error if we are not able to read from the configuration cache on the node.

### CODE REVIEW

**Comment by SchSeba** [comment]
followup on https://github.com/k8snetworkplumbingwg/sriov-network-operator/pull/734

@adrianchiris @ykulazhenkov please take a look :)

File: pkg/plugins/mellanox/mellanox_plugin.go [hunk 1]
```diff
@@ -194,36 +206,37 @@ func (p *MellanoxPlugin) Apply() error {
...
 	for _, iface := range nicPortsMap {
 		_, exist, err := p.helpers.LoadPfsStatus(iface.PciAddress)
 		if err != nil {
 			log.Log.Error(err, "failed to load PF status from disk", "address", iface.PciAddress)
-			continue
+			return false, err
 		}
 		if exist {
 			log.Log.V(2).Info("PF configured by the operator", "interface", iface)
-			return true
```
Comment (adrianchiris) [inline pkg/plugins/mellanox/mellanox_plugin.go:225]: so this was the bug ?

i see you reversed the meaning of this function

File: pkg/plugins/mellanox/mellanox_plugin.go [hunk 2]
```diff
@@ -144,12 +144,24 @@ func (p *MellanoxPlugin) OnNodeStateChange(new *sriovnetworkv1.SriovNetworkNodeS
 		pciAddress := pciPrefix + "0"
 
 		// Skip devices not configured by the operator
-		if p.nicNotConfiguredByOperator(portsMap) {
+		isConfigured, err := p.nicConfiguredByOperator(portsMap)
+		if err != nil {
+			return false, false, err
```
Comment (adrianchiris) [inline pkg/plugins/mellanox/mellanox_plugin.go:149]: thats going to be a non-recoverable error right ? (also below at L160)

i guess if we get here then state file got corrupted. which "should not" happen

Comment (adrianchiris) [reply pkg/plugins/mellanox/mellanox_plugin.go:149]: what about first time running on the interface ? or after nodeState got re-created (we clearing the state file there) ?

this logic runs before generic plugin which eventually stores the pf status when it configures sriov interface

**Review by adrianchiris — APPROVED** [review]
overall LGTM, 

left some small comments

File: pkg/plugins/mellanox/mellanox_plugin.go (hunk 2 above)
Comment (SchSeba) [reply pkg/plugins/mellanox/mellanox_plugin.go:149]: That is right. 
Maybe add an error message and request the user to remove the file?

Comment (adrianchiris) [reply pkg/plugins/mellanox/mellanox_plugin.go:149]: +1

File: pkg/plugins/mellanox/mellanox_plugin.go (hunk 1 above)
Comment (SchSeba) [reply pkg/plugins/mellanox/mellanox_plugin.go:225]: done

File: pkg/plugins/mellanox/mellanox_plugin.go (hunk 2 above)
Comment (SchSeba) [reply pkg/plugins/mellanox/mellanox_plugin.go:150]: done I put the info in the function itself (for both of them)

**Review by adrianchiris — APPROVED** [review]

**Comment by SchSeba** [comment]
thanks @adrianchiris 

merging

**Pull request merged**